libs/s25main/ingameWindows/iwMilitaryBuilding.cpp

```diff
@@ -204,9 +238,41 @@ void iwMilitaryBuilding::Msg_ButtonClick(const unsigned ctrl_id)
             }
         }
         break;
-        case 10: // send home button (addon)
+        case 10:
+        {
+            if(gwv.GetWorld().GetGGS().getSelection(AddonId::MILITARY_CONTROL) == 1)
+            {
+                // Send the highest rank soldiers in this building home and get new soldiers
+                if(building->GetNumTroops() > 1)
+                {
+                    auto maxRank = building->GetTroops().back().GetRank();
+                    gcFactory.SetDesiredTroops(building->GetPos(), maxRank, 0);
+                    gcFactory.SetDesiredTroops(building->GetPos(), maxRank, building->GetMaxTroopsCt());
+                }
+            }
+            else
+            {
+                // Send all soldiers home except one
+                gcFactory.SetDesiredTroops(building->GetPos(), 0, 1);
+                for(unsigned rank = 1; rank < NUM_SOLDIER_RANKS; ++rank)
+                    gcFactory.SetDesiredTroops(building->GetPos(), rank, 0);
+            }
+        }
+        break;
+        case 11: case 16: case 21: case 26: case 31:
+        {
+            unsigned rank = (ctrl_id - 11) / 5;
```

Comment:
Style: If you don't intend to change the value mark the variable `const`, like:
```suggestion
            const unsigned rank = (ctrl_id - 11) / 5;
```